app/assets/javascripts/discourse/app/lib/draft-state-cache.js

```diff
@@ -0,0 +1,57 @@
+// In-memory draft state store keyed by draftKey. No HTTP calls.
+// Values: { hasDraft: boolean, postId?: number, action?: string }
+const _store = new Map();
```

Comment:
IMO this would be better as a service rather than a lib...you could have the same functions here but also stick all the appEvents stuff here to avoid cluttering up the button components.

```javascript
export default class TopicDraftsManager extends Service {
  @appEvents;

  setDraftFromTopic(topic) {
  }

  hasTopicDraft(topic) {
  }

  saveDraft(key, data) {
  }

  destroyDraft(key) {
  }

  subscribeToDraftEvents(callbackFn) {
  }

  unsubscribeFromDraftEvents(callbackFn) {
  }
}
```